File: blepro/src/main/java/com/lepu/blepro/ext/Ad5Data.java
package com.lepu.blepro.ext;

public class Ad5Data {
    private String sn;
    private int hr1;
    private int hr2;

    public String getSn() {
        return sn;
    }

    public void setSn(String sn) {
        this.sn = sn;
    }

    public int getHr1() {
        return hr1;
    }

    public void setHr1(int hr1) {
        this.hr1 = hr1;
    }

    public int getHr2() {
        return hr2;
    }

    public void setHr2(int hr2) {
        this.hr2 = hr2;
    }

    @Override
    public String toString() {
        return "Ad5Data{" +
                "sn='" + sn + '\'' +
                ", hr1=" + hr1 +
                ", hr2=" + hr2 +
                '}';
    }
}

File: blepro/src/main/java/com/lepu/blepro/ble/data/Ad5Data.kt
package com.lepu.blepro.ble.data

import com.lepu.blepro.utils.ByteUtils.byte2UInt
import com.lepu.blepro.utils.HexString.trimStr

class Ad5Data(val bytes: ByteArray) {

    var cmd = 0
    var sn = ""
    var hr1 = 0
    var hr2 = 0

    init {
        cmd = byte2UInt(bytes[2])
        sn = trimStr(String(bytes.copyOfRange(3, 10)))
        hr1 = byte2UInt(bytes[11])
        hr2 = byte2UInt(bytes[12])
    }

    override fun toString(): String {
        return """
            Ad5Data : 
            cmd : $cmd
            sn : $sn
            hr1 : $hr1
            hr2 : $hr2
        """.trimIndent()
    }

}